linera-core/src/worker.rs

```diff
@@ -753,29 +754,34 @@ where
         &self,
         chain_id: ChainId,
     ) -> Result<ChainActorEndpoint<StorageClient>, WorkerError> {
-        let mut chain_workers = self.chain_workers.lock().await;
-
-        if !chain_workers.contains(&chain_id) {
-            chain_workers.push(
+        let mut receiver = None;
+        let sender = self
+            .chain_workers
+            .lock()
+            .unwrap()
+            .get_or_insert(chain_id, || {
+                let (sender, receiver_) = mpsc::unbounded_channel();
+                receiver = Some(receiver_);
+                sender
+            })
+            .clone();
+
+        if let Some(receiver) = receiver {
```

Comment:
Nit: I'd rename `receive` to `new_receiver`, because I think it makes it clearer that the receiver was just created.

```suggestion
        let mut new_receiver = None;
        let sender = self
            .chain_workers
            .lock()
            .unwrap()
            .get_or_insert(chain_id, || {
                let (sender, receiver) = mpsc::unbounded_channel();
                new_receiver = Some(receiver);
                sender
            })
            .clone();

        if let Some(receiver) = new_receiver {
```

Reply:
Works for me :)